src/api-wrapper.ts

```diff
@@ -50,8 +54,35 @@ export async function createDeploymentFromInputs(
   const deployments = await deploymentRepository.list({ ids: deploymentIds, take: deploymentIds.length })
 
   const envIds = deployments.Items.map(d => d.EnvironmentId)
-  const envRepository = new EnvironmentRepository(client, parameters.space)
-  const environments = await envRepository.list({ ids: envIds, take: envIds.length })
+  let environments: ResourceCollection<DeploymentEnvironmentV2 | DeploymentEnvironment>
+  const environmentV1Repository = new EnvironmentRepository(client, parameters.space)
+  const environmentV2Repository = new EnvironmentV2Repository(client, parameters.space)
+
+  try {
+    environments = await environmentV2Repository.list({ ids: envIds, skip: 0, take: envIds.length })
+  } catch (error) {
+    // Catch cases in which GetEnvironmentsRequestV2 capability is toggled off or not available on Octopus Server version.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    if ((error as any)?.StatusCode === 404) {
+      client.info('List environments v2 endpoint may be unavailable. Checking v1 endpoint...')

```

Comment:
🤔 I'm wondering if we want this at info level so that it shows all the time (for Server that doesn't support the right version) or whether debug might be better as it's a bit of an implementation detail. What do you think?